modules/core/domain/entities/tenant/tenant.go

```diff
@@ -0,0 +1,86 @@
+package tenant
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type Tenant struct {
```

Comment:
Is there a particular reason Tenant isn't an interface like other domain entities?